### PR TITLE
Add draft contribution and CoC guides

### LINKED ISSUE #3: Add contribution guide
We should add a contribution guide for the project that explains how to help (and that covers everything, from just raising an issue, to fixing documentation all the way to writing super-charged GPU code).

There are a few examples of ones that were done for recent HSF projects:

- prmon: <https://github.com/HSF/prmon/blob/master/doc/CONTRIBUTING.md>
- acts: <https://github.com/acts-project/acts/blob/master/CONTRIBUTING.md>
- phoenix: <https://github.com/HSF/phoenix/blob/master/CONTRIBUTING.md>

### PR DESCRIPTION
**Pull request by graeme-a-stewart**
Code of conduct is from CERN, which seems to be perfect for out project - thanks to @paulgessinger for having textified it in Acts.

Contribution guide is also an adapted version from Acts (thanks!), so makes reference to things that we have not yet setup in AdePT (marked by **FIXME**), but does form a good basis for discussion.

Note that as well as C++ I added some draft guidance for Python and Markdown files in the project.

Closes #3

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+  - mention affected version(s)
+  - issue type: "Bug"
+  - a detailed description of the bug including a recipe on how to reproduce it and any hints which may help diagnosing the problem
+- feature requests
+  - issue type: "Improvement" or "New Feature"
+  - a detailed description of the feature request including possible use cases and benefits for other users
+
+## Make a contribution
+
+Anyone is welcome to contribute to AdePT. Below is a short description how to contribute. If you have any questions, feel free to ask us **FIXME add mailing list** for help or guidance.
```
Comment (agheata) [inline CONTRIBUTING.md:42]: Mailing list, CI notifications - few issues to discuss next...

File: .flake8 [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# SPDX-FileCopyrightText: 2020 CERN
+# SPDX-License-Identifier: Apache-2.0
+
+# Minimal flake8 tweak to be compatible with black's line length
+# of 88 characters 
+# https://black.readthedocs.io/en/stable/the_black_code_style.html#line-length
+
+[flake8]
```
Comment (agheata) [inline .flake8:8]: How is this enforced?

**Review by agheata — APPROVED** [review]
Very nice!

File: .flake8 (hunk 2 above)
Comment (graeme-a-stewart) [reply .flake8:8]: It should be easy (we do it already in HSF/prmon)

- Add a cmake target `flake8` that runs the test <https://github.com/HSF/prmon/blob/575d3a42de48ea9f0a59ba8d018bba37dd9f8ab1/cmake/python-format.cmake#L16>
- Add an action script that invokes this test <https://github.com/HSF/prmon/blob/master/.github/scripts/flake8.sh>
- Add an action that triggers the script <https://github.com/HSF/prmon/blob/master/.github/workflows/flake8.yml>

The prmon action uses a container and we should probably have a project standard container environment for these kinds of checks.

Comment (drbenmorgan) [reply .flake8:8]: I'd guess by a CI check, though am not sure how to implement (but I know spack do this on their repo).

**Review by drbenmorgan — APPROVED** [review]
All good for me!

**Pull request merged**